config/initializers/rack_middleware.rb

```diff
@@ -0,0 +1,30 @@
+require 'rack/default_content_type'
+
+# This is where we introduce custom middleware that interacts with Rack
+# and Rails to change how requests are handled.
+Rails.application.configure do
+
+  # This configures which paths do and do not require token authentication.
+  # Token authentication is optional for authn routes, and it's not applied at
+  # all to authentication, host factories, or static assets (e.g. images, CSS)
+  config.middleware.use Conjur::Rack::Authenticator,
+  optional: [
+    /^\/authn-[^\/]+\//,
+    /^\/authn\//,
+    /^\/public_keys\//
+  ],
+  except: [
+    /^\/authn-[^\/]+\/.*\/authenticate$/,
+    /^\/authn\/.*\/authenticate$/,
+    /^\/host_factories\/hosts$/,
+    /^\/assets\/.*/,
+    /^\/authenticators$/,
+    /^\/$/
+  ]
```

Comment:
Can we get some parens around this function call?

Reply:
I am concerned that we don't know exactly what this is doing.  It appears to be existing code that we're just moving so doesn't have to be a blocker here, but something so central, with serious security implications, should be well-understand by everyone on the team, and we should have an explanatory comment here or at least a link to the docs